opencensus/trace/ext/grpc/server_interceptor.py

```diff
@@ -11,96 +11,87 @@
+from opencensus.trace.ext.grpc import utils as grpc_utils
 from opencensus.trace.propagation import binary_format
 
 ATTRIBUTE_COMPONENT = 'COMPONENT'
 ATTRIBUTE_ERROR_NAME = 'ERROR_NAME'
 ATTRIBUTE_ERROR_MESSAGE = 'ERROR_MESSAGE'
 RECV_PREFIX = 'Recv'
 
-RpcRequestInfo = collections.namedtuple(
-    'RPCRequestInfo', ('request', 'context')
-)
-# exc_info is the three tuple defined at:
-# https://docs.python.org/3/library/sys.html#sys.exc_info
-RpcResponseInfo = collections.namedtuple(
-    'RPCCallbackInfo', ('request', 'context', 'response', 'exc_info')
-)
-
-
-class RpcMethodHandlerWrapper(object):
-    """Wraps a grpc RPCMethodHandler and records the variables about the
-     execution context and response
-    """
-
-    def __init__(
-        self, handler, pre_handler_callbacks=None, post_handler_callbacks=None
-    ):
-        """
-        :param handler: instance of RpcMethodHandler
-
-        :param pre_handler_callbacks: iterable of callbacks that accept an
-         instance of RpcRequestInfo that are called before the server handler
-
-        :param post_handler_callbacks: iterable of callbacks that accept an
-         instance of RpcResponseInfo that are called after the server
-         handler finishes execution
-        """
-        self.handler = handler
-        self._pre_handler_callbacks = pre_handler_callbacks or []
-        self._post_handler_callbacks = post_handler_callbacks or []
-
-    def proxy(self, prop_name):
-        def _wrapper(request, context, *args, **kwargs):
-            for callback in self._pre_handler_callbacks:
-                callback(RpcRequestInfo(request, context))
-            exc_info = (None, None, None)
-            response = None
-            try:
-                response = getattr(
-                    self.handler, prop_name
-                )(request, context, *args, **kwargs)
-            except Exception as e:
-                logging.exception(e)
-                exc_info = sys.exc_info()
-                raise
-            finally:
-                for callback in self._post_handler_callbacks:
-                    callback(RpcResponseInfo(
-                        request, context, response, exc_info)
-                    )
-            return response
-
-        return _wrapper
-
-    def __getattr__(self, item):
-        if item in (
-            'unary_unary', 'unary_stream', 'stream_unary', 'stream_stream'
-        ):
-            return self.proxy(item)
-        return getattr(self.handler, item)
-
 
 class OpenCensusServerInterceptor(grpc.ServerInterceptor):
     def __init__(self, sampler=None, exporter=None):
         self.sampler = sampler
         self.exporter = exporter
 
-    def _start_server_span(self, rpc_request_info):
-        metadata = rpc_request_info.context.invocation_metadata()
+    def intercept_service(self, continuation, handler_call_details):
+        def latency_wrapper(behavior, request_streaming, response_streaming):
```

Comment:
Does this name need to be the same as the grpc example? Probably change to trace_wrapper?